### DISCUSSION TITLE
Desktop application for concore

### DISCUSSION THREAD

**Discussion by Rahuljagwani** [ControlCore-Project/concore#70] [Ideas]
@pradeeban 
It may be trivial but I wanted to know if concore editor and concore can be bundled together as a desktop application separately for Windows, Linux, and Mac making it easy and efficient for users (as mentioned before most users are neuroscientists) to use. 
The only thing that has to be done if the desktop application will be available is to download it, login if necessary (in case of contributions to the parent repository), and perform any executions possible.
(It can be a future GSoc idea if possible)

**Comment by pradeeban** [comment]
@Rahuljagwani, if you remember, we (You, @parteekcoder, and I) discussed this in the potential approaches for the CONTROL-CORE integrated experience.

For that project, one option is to bundle the concore-editor and concore and provide an installer. But we (in that GitHub discussion) thought of going with the github-based concore-editor deployment for the sake of lightweightedness. But if this is a trivial task, we could provide an installer as you propose as an optional one. This should not replace the current github-based concore editor approach though, since we don't want a bloated installer as the only option. Some people just want the concore in their repository. Not the whole ecosystem.

I think your suggestion can be a part of GSoC 2023 integrated CONTROL-CORE experience, depending on the effort needed.

**Reply by parteekcoder** [reply]
> @Rahuljagwani, if you remember, we (You, @parteekcoder, and I) discussed this in the potential approaches for the CONTROL-CORE integrated experience.
> 
> For that project, one option is to bundle the concore-editor and concore and provide an installer. But we (in that GitHub discussion) thought of going with the github-based concore-editor deployment for the sake of lightweightedness. But if this is a trivial task, we could provide an installer as you propose as an optional one. This should not replace the current github-based concore editor approach though, since we don't want a bloated installer as the only option. Some people just want the concore in their repository. Not the whole ecosystem.
> 
> I think your suggestion can be a part of GSoC 2023 integrated CONTROL-CORE experience, depending on the effort needed.

yeah @pradeeban, I also think that this can be included in the `integrated CONTROL-CORE experience` as it is a 350hrs project and can be done in that. Without this task, integration will not be fully completed

**Reply by Rahuljagwani** [reply]
Understood @pradeeban, as @parteekcoder mentioned this whole stuff can be included in the `integrated CONTROL-CORE experience` as 350hrs are provided for it, this can be done.

**Comment by tmjoris** [comment]
Hello everyone I wish to contribute on Project number 10, creating java reference librabries for concore. But let me ask where should I start? Is it with the issues highlighted in the repo or in the code translation

**Reply by pradeeban** [reply]
Hi @tmjoris,
Welcome!
I would say a good way to start would be to look into the C++ implementation of concore (because, it was a later implementation, mimicking the code from its Python implementation to C++ implementation).

So, if you get an idea how it was "translated' from Python to C++, you should be able to do the same with Python to Java as well. This project will therefore require you to be comfortable with both Java and Python.

concore is largely developed in Python, with implementations in other languages too (C++ is one). So, your implementation will be a Java version of that.

**Reply by tmjoris** [reply]
Hello @pradeeban 
Thank you for your response and warm welcome.
I will do so and I'll hit you up if I have any further questions

**Reply by tmjoris** [reply]
Hello @pradeeban let me ask why does the python implementation have a lines of code at the beginning that kills the entire process when concore is run on a windows desktop?

**Reply by Rahuljagwani** [reply]
Hi @tmjoris, can you let me know the exact lines from the file? I remember implementing `conore` for Windows last year, although it is not completely implemented (Some things include docker for Windows, etc.). Feel free to reach out to me, if you need any kind of help.

Also, Hi @pradeeban :)

**Reply by tmjoris** [reply]
Hi @Rahuljagwani, in the file concore.py these line 11 to 14 :

if hasattr(sys, 'getwindowsversion'):
    with open("concorekill.bat","w") as fpid:
        fpid.write("taskkill /F /PID "+str(os.getpid())+"\n")

appears to create a bat file concore.bat which is written into a command which kills the process. Please correct me if my analysis is incorrect.
